web-new/src/pages/ConfigEditor.tsx

```diff
@@ -1,10 +1,151 @@
+import useSWR from "swr";
+import * as monaco from "monaco-editor";
+import { configureMonacoYaml } from "monaco-yaml";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { useApiHost } from "@/api";
 import Heading from "@/components/ui/heading";
+import ActivityIndicator from "@/components/ui/activity-indicator";
+import { Button } from "@/components/ui/button";
+import axios from "axios";
+import copy from "copy-to-clipboard";
+
+type SaveOptions = "saveonly" | "restart";
 
 function ConfigEditor() {
+  const apiHost = useApiHost();
+  const { data: config } = useSWR<string>("config/raw");
+  const [success, setSuccess] = useState<string | undefined>();
+  const [error, setError] = useState<string | undefined>();
+  const editorRef = useRef<any | null>();
```

Comment:
There is probably a type you can use here, maybe something like `<monaco.editor.IStandaloneCodeEditor | null>`?